File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var maxChunksToSorted = function(arr) {
    const n = arr.length;
    let ans = 0, maxVal = 0;

    for(let i = 0; i<n; i++) {
        maxVal = Math.max(maxVal, arr[i]);
        if(maxVal == i) ans++;
    }

    return ans;
};

File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.java
class Solution {
    public int maxChunksToSorted(int[] arr) {
        int n = arr.length;
        int ans = 0, maxVal = 0;

        for(int i = 0; i<n; i++) {
            maxVal = Math.max(maxVal, arr[i]);
            if(maxVal == i) ans++;
        }

        return ans;
    }
}